### PR TITLE
R2frida documentation for 0x05b

### PR DESCRIPTION
**Pull request by cldrn**
This PR covers issue #1232.

### CODE REVIEW

File: Document/0x05b-Basic-Security_Testing.md [hunk 1]
```diff
@@ -281,7 +281,120 @@ Frida also provides bindings for various languages, including Python, C, NodeJS,
 
 ##### r2frida
 
--- ToDo: <https://github.com/OWASP/owasp-mstg/issues/1232>
+[r2frida](https://github.com/nowsecure/r2frida "r2frida on Github") is a project that allows radare2 to connect to Frida, effectively merging the powerful reverse engineering capabilities of radare2 with the dynamic instrumentation toolkit of Frida. R2frida allows you to:
+
+- Attach radare2 to any local process or remote frida-server via USB or TCP.
+- Read/Write memory from the target process.
+- Load Frida information such as maps, symbols, imports, classes and methods into radare2.
+- Call r2 commands from frida as it exposes the r2pipe interface into the Frida Javascript API.
```
Comment (sushi2k) [inline Document/0x05b-Basic-Security_Testing.md]: frida -> Frida

File: Document/0x05b-Basic-Security_Testing.md [hunk 2]
```diff
@@ -281,7 +281,120 @@ Frida also provides bindings for various languages, including Python, C, NodeJS,
...
+
+However, you may also compile r2frida yourself in the traditional way. You can find how to do this in the official documentation of [r2frida](https://github.com/nowsecure/r2frida/blob/master/README.md "r2frida readme").
+
+###### Using r2frida
+
+With frida-server running, you should now be able to attach to it using the pid, spawn path, host and port and USBID/pid. For example, to attach to PID 1234:
+
+```shell
+$ r2 frida://1234
+```
```
Comment (sushi2k) [inline Document/0x05b-Basic-Security_Testing.md]: Maybe you can add:

For more examples see the [usage section in the r2frida repo](https://github.com/nowsecure/r2frida/blob/master/README.md#usage "Usage of r2frida")

File: Document/0x05b-Basic-Security_Testing.md [hunk 3]
```diff
@@ -281,7 +281,120 @@ Frida also provides bindings for various languages, including Python, C, NodeJS,
...
+
+```shell
+$ r2pm -ci r2frida
+```
+
+However, you may also compile r2frida yourself in the traditional way. You can find how to do this in the official documentation of [r2frida](https://github.com/nowsecure/r2frida/blob/master/README.md "r2frida readme").
+
+###### Using r2frida
+
+With frida-server running, you should now be able to attach to it using the pid, spawn path, host and port and USBID/pid. For example, to attach to PID 1234:
```
Comment (sushi2k) [inline Document/0x05b-Basic-Security_Testing.md]: USBID/pid -> device-id

File: Document/0x05b-Basic-Security_Testing.md [hunk 4]
```diff
@@ -281,7 +281,120 @@ Frida also provides bindings for various languages, including Python, C, NodeJS,
...
+Searching 12 bytes: 75 6e 61 63 63 65 70 74 61 62 6c 65
+Searching 12 bytes in [0x0000561f05ebf000-0x0000561f05eca000]
+...
+Searching 12 bytes in [0xffffffffff600000-0xffffffffff601000]
+hits: 23
+{"address":"0x561f072c4223","size":12,"flag":"hit14_1","content":"unacceptable policyunsupported md algorithmvar bad valuec0"},{"address":"0x561f072c4275","size":12,"flag":"hit14_2","content":"unacceptableSearching 12 bytes: 75 6e 61 63 63 65 70 74 61"},{"address":"0x561f072c42c8","size":12,"flag":"hit14_3","content":"unacceptableSearching 12 bytes: 75 6e 61 63 63 65 70 74 61 "},
+...
+```
+
+To list the loaded libraries and filter the results using the internal grep from radare2:
```
Comment (sushi2k) [inline Document/0x05b-Basic-Security_Testing.md]: can you explain the command used below a bit further in the sentence above?

File: Document/0x05b-Basic-Security_Testing.md [hunk 5]
```diff
@@ -281,7 +281,120 @@ Frida also provides bindings for various languages, including Python, C, NodeJS,
...
+[r2frida](https://github.com/nowsecure/r2frida "r2frida on Github") is a project that allows radare2 to connect to Frida, effectively merging the powerful reverse engineering capabilities of radare2 with the dynamic instrumentation toolkit of Frida. R2frida allows you to:
+
+- Attach radare2 to any local process or remote frida-server via USB or TCP.
+- Read/Write memory from the target process.
+- Load Frida information such as maps, symbols, imports, classes and methods into radare2.
+- Call r2 commands from frida as it exposes the r2pipe interface into the Frida Javascript API.
+
+###### Installing r2frida
+
+Before installing r2frida, make sure you have installed radare2 on your system. In GNU/Debian you also need to install the following dependencies:
```
[the remark this answers is not in the document]
Comment (TheDauntless) [reply Document/0x05b-Basic-Security_Testing.md]: In GNU -> On GNU

**Comment by commjoen** [comment]
Thanks!

**Pull request merged**